### PR TITLE
A new era of Gunk™: Biomass tator item.

### PR DESCRIPTION
**Pull request by Eneocho**
Have you EVER thought to yourself: 
**_This station is so hard to gunk™! If only there was an easier way!_**
image
Hi, I'm **Derek Baum VII** and I'm declaring **_WAR_** on clean stations with the new _Biomass Packet_. Just buy, shake it around, throw it in a station, and let the Biomass Packet do the rest. **BANG! AND THE GUNK IS DONE!**
## What this does
Syndicate agents can purchase a biomass sample packet for a neat and tidy 8 TC, letting them release The Gunk™ on any unsuspecting station. Due to their essences being bound to The Gunk™ Assistants and Janitors only need 5TC for a sample.
Just like a powercreep packet, just shake and throw and it'll release the biomass.
## Why it's good
Biomass cleanups tend to involve a good chunk of the crew, which usually leaves areas unattended during the cleanup. Perfect for a would-be entrepreneur to enter and steal anything they need.
:cl:
 * rscadd: New Biomass Sample Packet™, for the Gunk™ Enthusiast™ on the go™. Available for 5TC for Assistants and Janitors, 8TC for everyone else.
 * imageadd: Biomass sample packet icon.

### CODE REVIEW

**Comment by ghost** [comment]
why so expensive?

**Comment by Eneocho** [comment]
Powercreep costs 16 and 10 with discount, this is fairly cheaper at 10 and 5 (though you could argue powercreep should be made cheaper as well, since distributing gloves neuters it completely).

**Comment by pedr69** [comment]
with gloves, powercreep is harmless and actually benefits the station - giving power into the grid
the biomass stops movement and is WAY slower to clear, plus has a tiny hitbox

**Comment by Cartlord** [comment]
hitbox can be made irrelevant by using the alt-click menu

**Comment by ManopolisJones** [comment]
I still think we need biomass that shoots a random reagent at you while you're adjacent to it

**Comment by Eneocho** [comment]
>I still think we need biomass that shoots a random reagent at you while you're adjacent to it

Sounds like a !!!FUN!!! idea but would make it nigh impossible to remove.

**Comment by Eneocho** [comment]
Thinking it further and testing on localhost it feels a bit more fair to have it cost 8TC for non jobs than a full 10TC, considering the relative harmlessness of the biomass. If anyone feels it should be even cheaper or more expensive, please tell.

**Comment by Blithering** [comment]
How about the weird purple balls popping on destruction, spreading unnamed gunk around? Really get under the janitors skin.

**Comment by luigil101** [comment]
> > I still think we need biomass that shoots a random reagent at you while you're adjacent to it
> 
> Sounds like a !!!FUN!!! idea but would make it nigh impossible to remove.

Not if you use a ripley to clear it
in an incredibly rare turn of events, using a mech is actually _faster_ than doing it by hand

File: code/game/gamemodes/events/biomass.dm [hunk 1]
```diff
@@ -205,3 +205,34 @@
 		new/obj/effect/biomass_controller(Floor) // spawn a controller at floor
 		log_admin("Event: Biomass spawned at [Floor.loc] ([Floor.x],[Floor.y],[Floor.z]).")
 		message_admins("<span class='notice'>Event: Biomass spawned at [Floor.loc] <A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[Floor.x];Y=[Floor.y];Z=[Floor.z]'>(JMP)</a></span>")
+
+//biomass tator item, code "borrowed" from powercreeper_packet.dm
```
Comment (Kurfursten) [inline code/game/gamemodes/events/biomass.dm]: instead why not just make them one in the same as some sort of deployable packet item? the only difference is the icon and name, seems a perfect time for inheritance

Comment (Eneocho) [reply code/game/gamemodes/events/biomass.dm]: I tried implementing it as a child of the powercreeper packet, but it didn't want to spawn biomass. There's also a small difference in the examine text (the biomass one has some flavour too). I can try to rework it a bit however.

Comment (Eneocho) [reply code/game/gamemodes/events/biomass.dm]:
```
/obj/item/deployable_packet/
	desc = ""
	w_class = W_CLASS_TINY
	var/activated = 0
	var/deployeditem = null

/obj/item/deployable_packet/attack_self(mob/user)
	if(!istype(user))
		return
	if(activated)
		return

	to_chat(user, "<span class='warning'>You shake \the [src].</span>")
	to_chat(user, "<span class='danger'>It starts moving weirdly!</span>")
	activated = 1

	spawn(3 SECONDS)
		new deployeditem(get_turf(src))
		qdel(src)

/obj/item/deployable_packet/biomass
	name = "Biomass Packet"
	icon = 'icons/obj/biomass.dmi'
	icon_state = "packet"
	deployeditem = /obj/effect/biomass_controller

/obj/item/deployable_packet/powercreeper/examine(mob/user, size, show_name)
	. = ..()
	to_chat(user, "Derek Baum VII's latest innovation. Carries a cryo-stabilized sample of biomass tissue.")
	to_chat(user, "It reads:\nStep 1: Shake to active.\nStep 2: Throw.\nStep 3: Enjoy.")
	if(activated)
		to_chat(user, "<span class='danger'>It's bubbling weirdly!</span>")
			
/obj/item/deployable_packet/powercreeper
	name = "powercreeper Packet"
	icon = 'icons/obj/structures/powercreeper.dmi'
	icon_state = "packet"
	deployeditem = /obj/structure/cable/powercreeper
	
/obj/item/deployable_packet/powercreeper/examine(mob/user, size, show_name)
	. = ..()
	to_chat(user, "Derek Baum VII's favourite creation. Carries a self-replicating sample of powercreeper.")
	to_chat(user, "It reads:\nStep 1: Shake to active.\nStep 2: You have 3 seconds to run.")
	if(activated)
		to_chat(user, "<span class='danger'>It's sizzling weirdly!</span>")

```

Something like this?

Comment (Kurfursten) [reply code/game/gamemodes/events/biomass.dm]: Yes, that would work, although you made a typo in the path of the examine for biomass which would override powercreeper's examine

Comment (Eneocho) [reply code/game/gamemodes/events/biomass.dm]: Alright, I'll implement it now then (and thanks for noticing that typo).

Comment (Eneocho) [reply code/game/gamemodes/events/biomass.dm]: Would it be better to have it as its own file (deployable packet.dm)?

Comment (Kurfursten) [reply code/game/gamemodes/events/biomass.dm]: since powercreeper_packet was its own file I would delete that one and make a new one named as you say

Comment (Eneocho) [reply code/game/gamemodes/events/biomass.dm]: Alright, I think that should be it. Please check.

Comment (Eneocho) [reply code/game/gamemodes/events/biomass.dm]: I had a brainblast for space vine packets, but that's for another PR.

Comment (Eneocho) [reply code/game/gamemodes/events/biomass.dm]: @Kurfursten

**Pull request merged**